lib/senders/oauth_client_info.js

```diff
@@ -31,58 +25,37 @@ module.exports = (log, config) => {
    * @param clientId
    * @returns {Promise<any>}
    */
-  function fetch(clientId) {
+  async function fetch(clientId) {
     log.trace({ op: 'fetch.start' })
 
-    const options = {
-      url: `${OAUTH_SERVER}/v1/client/${clientId}`,
-      method: 'GET',
-      json: true
+    if (! clientId || clientId === 'sync') {
+      log.trace({ op: 'fetch.sync' })
+      return FIREFOX_CLIENT
     }
 
-    return new P((resolve) => {
-      if (! clientId || clientId === 'sync') {
-        log.trace({ op: 'fetch.sync' })
-        return resolve(FIREFOX_CLIENT)
-      }
-
-      clientCache.get(clientId)
-        .then((cachedRecord) => {
-          if (cachedRecord) {
-            // used the cachedRecord
-            log.trace({ op: 'fetch.usedCache' })
-            return resolve(cachedRecord)
-          }
-
-          // request info from the OAuth server
-          request(options, function (err, res, body) {
-            if (err || res.statusCode !== 200 || ! body.name) {
-              if (err) {
-                log.critical({ op: 'fetch.failed', err: err })
-              } else {
-                log.warn({ op: 'fetch.failedForClient', clientId: clientId, name: !! body.name })
-              }
-              // fallback to the Firefox client if request fails
-              return resolve(FIREFOX_CLIENT)
-            }
-
-            const validation = Joi.validate(body.name, Joi.string().max(256).regex(DISPLAY_SAFE_UNICODE).required())
-            if (validation.error) {
-              // fallback to the Firefox client if invalid name
-              return resolve(FIREFOX_CLIENT)
-            }
+    const cachedRecord = await clientCache.get(clientId)
+    if (cachedRecord) {
+      // used the cachedRecord if it exists
+      log.trace({ op: 'fetch.usedCache' })
+      return cachedRecord
+    }
 
-            log.trace({ op: 'fetch.usedServer', body: body })
-            const clientInfo = {
-              // only providing `name` for now
-              name: body.name
-            }
-            resolve(clientInfo)
-            clientCache.set(clientId, clientInfo)
-          })
+    let clientInfo
+    try {
+      clientInfo = await oauthdb.getClientInfo(clientId)
+    } catch (err) {
+      // fallback to the Firefox client if request fails
+      if (! err.statusCode) {
+        log.critical({ op: 'fetch.failed', err: err })
+      } else {
+        log.warn({ op: 'fetch.failedForClient', clientId })
+      }
+      return FIREFOX_CLIENT
+    }
 
-        })
-    })
+    log.trace({ op: 'fetch.usedServer', body: clientInfo })
+    await clientCache.set(clientId, clientInfo)
+    return clientInfo
```

Comment:
behaviour change here? before we never waited for `.set` to complete and resolved before it